docs/configuring-playbook-synapse.md

```diff
@@ -24,7 +24,23 @@ If you plan to rely only on SSO and don't want your users to ever login with pas
 matrix_synapse_password_config_enabled: true
 ```
 
-Refer to [official doc](https://github.com/matrix-org/synapse/blob/develop/docs/openid.md) for examples.
+Add this to allow seamless forwarding to element web app and element android app. Without this setting matrix will ask the user if he trusts the app he tries to login.  
+
+```yaml
+matrix_synapse_sso:
```

Comment:
I don't see `matrix_synapse_sso` being defined and used anywhere. Is something missing from your branch?

Reply:
matrix_synapse_sso is used in `matrix-synapse/templates/synapse/homeserver.yaml`:
https://github.com/eraga/matrix-docker-ansible-deploy/blob/master/roles/matrix-synapse/templates/synapse/homeserver.yaml.j2#L1855

and in `matrix-synapse/defaults.yml`: 
https://github.com/eraga/matrix-docker-ansible-deploy/blob/master/roles/matrix-synapse/defaults/main.yml#L466

The last one should be fixed though (it is misconfigured), gimme a minute, I'll push another commit.

Reply:
done

Reply:
Why isn't this converted to something more flat?

```yaml
matrix_synapse_sso:
  client_whitelist: []
```

Seems like there's no point to having a multi-level object and then having to reach into it?

Reply:
To keep same structure as synapse has.

Reply:
This makes it harder for us to introduce new defaults that the user can use as-is, forcing them to always redefine the whole `matrix_synapse_sso` object.

For example, we can define: `matrix_synapse_sso_client_whitelist` and put a default value of empty in the role, and a new default in `group_vars/matrix_servers` like this:

```yaml

matrix_synapse_sso_client_whitelist: |
  {{
    []
    +
    ([matrix_server_fqn_element] if matrix_client_element_enabled else [])
  }}
```

If Synapse adds more keys under `sso`, we would be able to support them using variables like `matrix_sso_*`, and the `matrix_synapse_sso` object won't get in the way.

Reply:
understood. I'll correct it

Reply:
ok, done :)